src/Core/ContainerInterface.php

```diff
@@ -31,33 +30,6 @@
  * @experimental This interface is not final and can have additional
  *   methods and parameters added in future minor releases.
  */
-interface ContainerInterface extends PsrInterface
+interface ContainerInterface extends DefinitionContainerInterface
```

Comment:
Making `Cake\Core\ContainerInterface` extend `League\Container\DefinitionContainerInterface` was the only sane way I could think of to satisfy the typing requirements.